app/code/Magento/InventorySales/Plugin/Catalog/Model/Product.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Magento\InventorySales\Plugin\Catalog\Model;
+
+use Magento\Catalog\Model\Product as Subject;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\IsProductSalableInterface;
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+
+/**
+ * Check if product is salable in MSI
+ */
+class Product
+{
+    /**
+     * @var IsProductSalableInterface
+     */
+    private $isProductSalable;
+
+    /**
+     * @var StockResolverInterface
+     */
+    private $stockResolver;
+
+    /**
+     * @param IsProductSalableInterface $isProductSalable
+     * @param StockResolverInterface $stockResolver
+     */
+    public function __construct(
+        IsProductSalableInterface $isProductSalable,
+        StockResolverInterface $stockResolver
+    ) {
+        $this->isProductSalable = $isProductSalable;
+        $this->stockResolver = $stockResolver;
+    }
+
+    /**
+     * @param Subject $subject
+     * @param bool $result
+     */
+    public function afterIsSalable(Subject $subject, $result)
+    {
+        if (!$result) {
```

Comment:
I believe we should ignore original method behavior since it can be inconsistent with MSI enabled. 
Looks like test failures indicate the issue with it. Please, remove this check and try to use MSI logic only.

Reply:
so you want an around plugin ?

Reply:
sorry, that my progress is that slow, i was "out of time" in this period